app/code/Magento/SalesRule/Test/Mftf/Test/StorefrontCartFixedAmountWholeCartDiscountForBundleProductTest.xml

```diff
@@ -0,0 +1,146 @@
+            <actionGroup ref="DeleteCartPriceRuleByName" stepKey="DeleteCartPriceRuleByName">
+                <argument name="ruleName" value="{{SimpleSalesRule.name}}"/>
+            </actionGroup>
+
+            <!-- Delete Bundle product and it's children -->
+            <deleteData createDataKey="createBundleProduct" stepKey="createBundleProduct"/>
+            <deleteData createDataKey="simpleProduct1" stepKey="simpleProduct1"/>
+            <deleteData createDataKey="simpleProduct2" stepKey="simpleProduct2"/>
+            <deleteData createDataKey="simpleProduct3" stepKey="simpleProduct3"/>
+            <deleteData createDataKey="simpleProduct4" stepKey="simpleProduct4"/>
+
+            <!-- Revert Attribute 'sku' to it's default value (not accessible for Promo Rule Conditions) -->
+            <actionGroup ref="NavigateToEditProductAttributeActionGroup" stepKey="editSkuAttribute">
+                <argument name="ProductAttribute" value="sku"/>
+            </actionGroup>
+            <actionGroup ref="ChangeUseForPromoRuleConditionsProductAttributeActionGroup" stepKey="changeAttributePromoRule">
+                <argument name="option" value="0"/>
+            </actionGroup>
+
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+
+            <!-- Reindex invalidated indices after product attribute has been created/deleted -->
+            <magentoCron groups="index" stepKey="reindexInvalidatedIndices2"/>
+        </after>
+
+        <!-- Create a cart price rule for Fixed amount discount for whole cart -->
+        <actionGroup ref="AdminOpenCartPriceRulesPageActionGroup" stepKey="amOnCartPriceList"/>
+        <click selector="{{AdminCartPriceRulesSection.addNewRuleButton}}" stepKey="clickAddNewRule"/>
+        <fillField selector="{{AdminCartPriceRulesFormSection.ruleName}}" userInput="{{SimpleSalesRule.name}}" stepKey="fillRuleName"/>
+        <selectOption selector="{{AdminCartPriceRulesFormSection.websites}}" userInput="Main Website" stepKey="selectWebsites"/>
+        <selectOption selector="{{AdminCartPriceRulesFormSection.customerGroups}}" userInput="NOT LOGGED IN" stepKey="selectCustomerGroup"/>
+        <generateDate date="-1 day" format="m/d/Y" stepKey="yesterdayDate"/>
+        <fillField selector="{{AdminCartPriceRulesFormSection.fromDate}}" userInput="{$yesterdayDate}" stepKey="fillFromDate"/>
+        <click selector="{{AdminCartPriceRulesFormSection.actionsHeader}}" stepKey="clickToExpandActions"/>
+        <selectOption selector="{{AdminCartPriceRulesFormSection.apply}}" userInput="Fixed amount discount for whole cart" stepKey="selectActionType"/>
+        <fillField selector="{{AdminCartPriceRulesFormSection.discountAmount}}" userInput="4.99" stepKey="fillDiscountAmount"/>
+        <click selector="{{AdminCartPriceRulesFormSection.save}}" stepKey="clickSaveButton"/>
+        <see selector="{{AdminCartPriceRulesSection.messages}}" userInput="You saved the rule." stepKey="seeSuccessMessage"/>
```

Comment:
I'd create a new ActionGroup that is going to create a new Cart Rule, based on our needs. As arguments, you'll have all the hardcoded inputs from this test:
- ruleName
- websites
- customerGroups
- apply
- discountAmount

This way, we'll be able to re-use this action group in some other tests.
Thank you.